### PR TITLE
Issue 332 support "DROP TEMPORARY table"

### PR DESCRIPTION
**Pull request by gleonid**
### What problem does this PR solve? 
issue https://github.com/pingcap/parser/issues/332
adding support for `DROP TEMPORARY TABLE` syntax

### What is changed and how it works?
a. grammar has changed to support optional TEMPORARY keyword in DROP TABLE statement
b. `ast.DropTableStmt` has an extra boolean member

Tests 

 - Unit test
**added**

Code changes
 - Has exported variable/fields change
new field (`IsTemporary`) has been added to `ast.DropTableStmt`

Side effects
 - Breaking backward compatibility
all existing tests pass

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @winkyao

**Comment by gleonid** [comment]
@winkyao 
Can you please decide faith of this PR

I can resolve conflict on `parser.go` (generated file), but most likely it will become a conflict again as soon as any change to grammar is made (another PR merged).

I have `parser.go` as a separate commit, specifically for purpose to simplify rebasing.
I am happy to do at any point, or you can do it as easily.

Please let me know how you'd like to proceed

**Comment by gleonid** [comment]
conflict resolved. build is clean
please approve or decline this PR

File: parser.y [hunk 1]
```diff
@@ -2836,6 +2836,15 @@ DropTableStmt:
 	{
 		$$ = &ast.DropTableStmt{IfExists: true, Tables: $5.([]*ast.TableName), IsView: false}
 	}
+|	"DROP" "TEMPORARY" TableOrTables TableNameList RestrictOrCascadeOpt
```
Comment (bb7133) [inline parser.y]: Can we have an `OptTemporary` just like MySQL?

https://github.com/mysql/mysql-server/blob/124c7ab1d6f914637521fd4463a993aa73403513/sql/sql_yacc.yy#L11702

Comment (gleonid) [reply parser.y]: done.
And also using `IfExists`

**Comment by winkyao** [comment]
thanks for your contribution.

**Pull request merged**